app/src/main/java/com/android/sample/model/fridge/localData/RoomFridgeItemRepository.kt

```diff
@@ -0,0 +1,31 @@
+package com.android.sample.model.fridge.localData
+
+import com.android.sample.model.fridge.FridgeItem
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+class RoomFridgeItemRepository(
+    private val fridgeItemDao: FridgeItemDao,
+    private val dispatcher: CoroutineDispatcher
+) : FridgeItemLocalRepository {
+
+  override fun add(fridgeItem: FridgeItem) {
+    CoroutineScope(dispatcher).launch { fridgeItemDao.insert(fridgeItem.toFridgeItemEntity()) }
+  }
+
+  override fun delete(fridgeItem: FridgeItem) {
+    CoroutineScope(dispatcher).launch { fridgeItemDao.delete(fridgeItem.toFridgeItemEntity()) }
+  }
+
+  override fun getAll(onSuccess: (List<FridgeItem>) -> Unit, onFailure: (Exception) -> Unit) {
+    CoroutineScope(dispatcher).launch {
+      try {
+        val fridgeItems = fridgeItemDao.getAll().map { it.toFridgeItem() }
+        onSuccess(fridgeItems)
+      } catch (e: Exception) {
+        onFailure(e)
+      }
```

Comment:
Honestly, I'm passing the dispatcher as an argument because SonarCloud flagged it as an issue otherwise, citing maintainability concerns. To address this, I'm passing `Dispatchers.IO` as a parameter, as it is appropriate for writing to the device's storage.